include/bsp/iotc_bsp_io_fs.h

```diff
@@ -32,223 +29,175 @@
- * then passes resource_handle_out to <code>iotc_bsp_io_fs_read()</code> or
- * <code>iotc_bsp_io_fs_write()</code> to perform the corresponding operation.
- *
- * @param [in] resource_name File name.
- * @param [in] size (Optional) The size, in bytes, of the file. Required on
- *     non-POSIX implementations.
- * @param [in] open_flags An iotc_bsp_io_fs_open_flags_t bitmask.
- * @param [out] resource_handle_out A pointer to an
- *     iotc_bsp_io_fs_resource_handle handle.
- *
- * @see iotc_bsp_io_fs_read
- * @see iotc_bsp_io_fs_write
- * @see iotc_bsp_io_fs_state_e
- * @see iotc_bsp_io_fs_open_flags_t
- * @see iotc_bsp_io_fs_resource_handle_t
- *
- * @return iotc_bsp_io_fs_state_t File management function status.
+ * @details Opens a file.
+ *
+ * @param [in] resource_name The filename.
+ * @param [in] size The size, in bytes, of the file.
+ * @param [in] open_flags A {@link ::iotc_bsp_io_fs_open_flags_t file operation}
+ *     bitmask.
+ * @param [out] resource_handle_out A
+ *     {@link ::iotc_bsp_io_fs_resource_handle_t handle to an open file}.
  */
 iotc_bsp_io_fs_state_t iotc_bsp_io_fs_open(
     const char* const resource_name, const size_t size,
     const iotc_bsp_io_fs_open_flags_t open_flags,
     iotc_bsp_io_fs_resource_handle_t* resource_handle_out);
 
 /**
- * @function
- * @brief Read a file.
- *
- * The Device SDK passes an iotc_bsp_io_fs_resource_handle_t handle to
- * this function in order to read a file. Before calling this function,
- * the Device SDK calls the <code>iotc_bsp_io_fs_open()</code> function
- * to create an iotc_bsp_io_fs_resource_handle_t handle.
+ * @brief Reads a file.
  *
- * Implementations of this function must fill the buffer parameter at
- * offset 0.
+ * @details The function must fill the buffer at offset 0. The function can
+ * allocate buffers by:
+ *     - Allocating the buffer once, resusing it at each function call,
+ *       and freeing it when the file closes.
+ *     - Creating a new buffer each time the function is called and
+ *       and freeing the old buffer before the function returns.
  *
- * Implementations of this function can allocate a buffer once, reuse the
- * buffer, and free the buffer when the file is closed. Or, implementations can
- * create a new buffer on repeated calls and free the previous one before
- * returning from this function.
- *
- * @param [in] resource_handle The iotc_bsp_io_fs_resource_handle_t handle from
- *     <code>iotc_bsp_io_fs_open()</code>.
+ * @param [in] resource_handle A
+ *     {@link ::iotc_bsp_io_fs_resource_handle_t handle to an open file}.
  * @param [in] offset The position within the resource, in bytes, from which
  *     to start read operations.
  * @param [out] buffer A pointer to a buffer with the bytes read from the file.
- *     The buffer is already allocated by the Device SDK.
+ *     Function implementations must define how to allocate buffers The buffer
```

Comment:
"Function implementations must define how to allocate buffers" Where did this come from? Maybe I don't understand it, but it is obvious that an impl defines the allocation, since it has to implement it. So I'd rather remove this sentence.

Reply:
Done